pkg/db/errors.go

```diff
@@ -18,3 +20,22 @@ func isDialError(err error) bool {
 	netError := &net.OpError{}
 	return errors.As(err, &netError) && netError.Op == "dial"
 }
+
+func handleSQLError(query string, err error, log logging.Logger, cmdType string) error {
+	if err == nil {
+		return nil
+	}
```

Comment:
1. can be method of `dbTx` - don't need to pass the logger in this case
2. no need to check the err - I think you need to call this func as part of returning an error
3. would change the method signature to accept the (type, query and err) or (err, type, query) - find it hard in the current order to find a meaning (maybe I should not)